src/RestaurantContainer.jsx

```diff
@@ -2,46 +2,27 @@ import { useEffect } from 'react';
 import { useParams } from 'react-router-dom';
 import { useSelector, useDispatch } from 'react-redux';
 
+import Loading from './Loading';
+import Restaurant from './Restaurant';
+
 import { loadRestaurant } from './actions';
 
 import { get } from './utils';
 
 export default function RestaurantContainer() {
   const { id } = useParams();
-  const { name, address, menuItems } = useSelector(get('restaurant'));
-  const { loading } = useSelector((state) => ({
-    loading: state.loading,
-  }));
+  const restaurant = useSelector(get('restaurant'));
+  const loading = useSelector(get('loading'));
 
   const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch(loadRestaurant(id));
   }, []);
 
-  const loadedRender = (
+  return (
```

Comment:
guard clauses 로 처리해주시는게 더 깔끔할거같습니다 :)
```suggestion
    if(loading) {
      return (
        <Loading />
      )
    }
    
  return (
    <Restaurant restaurant={restaurant} />
  )
```

Reply:
넵, return JSX 안에서 if문을 사용할 수 없는 것으로 알고 있어서, guard clause로 loading 확인하면 Loading 리턴, 아니면 Restaurant 리턴으로 수정하였습니다.

```
  if (!restaurant.id) return <NotFoundPage />;

  return (
    <>
      <h2>{name}</h2>
      <p>
        주소:
        {address}
      </p>
      <h3>메뉴</h3>
      <ul>
        {menuItems && menuItems.map((menu) => (
          <li key={menu.id}>{menu.name}</li>
        ))}
      </ul>
    </>
  );
```